lib/SILGen/SILGenBridging.cpp

```diff
@@ -1558,22 +1557,6 @@ void SILGenFunction::emitNativeToForeignThunk(SILDeclRef thunk) {
   auto loc = thunk.getAsRegularLocation();
   loc.markAutoGenerated();
   Scope scope(Cleanups, CleanupLocation(loc));
-  
-  // Hop to the actor for the method's actor constraint, if any.
-  // Note that, since an async native-to-foreign thunk only ever runs in a
-  // task purpose-built for running the Swift async code triggering the
-  // completion handler, there is no need for us to hop back to the existing
-  // executor, since the task will end after we invoke the completion handler.
-  if (F.isAsync()) {
```

Comment:
I think we still need to do this hop if the function we're calling is synchronous.

Reply:
Unfortunately we can't do that, because the ObjC thunk would then be synchronous and not have a completion-handler.